doc/compiling/windows.md

```diff
@@ -2,67 +2,82 @@
-- `gmp` and `jsoncpp` are optional, otherwise the bundled versions will be compiled
-- `gettext` is optional, but required to use translations.
+This command takes about 10-30 minutes to complete, depending on your device.
 
-There are other optional libraries, but they are not tested if they can build and link correctly.
+-   `curl` is optional, but required to read the serverlist, `curl[winssl]` is required to use the content store.
+-   `openal-soft`, `libvorbis` and `libogg` are optional, but required to use sound.
+-   `luajit` is optional, it replaces the integrated Lua interpreter with a faster just-in-time interpreter.
+-   `gmp` and `jsoncpp` are optional, otherwise the bundled versions will be compiled
+-   `gettext` is optional, but required to use translations.
 
-Use `--triplet` to specify the target triplet, e.g. `x64-windows` or `x86-windows`.
+There are other optional libraries, but we don't test if they can build and link correctly.
 
+Use `--triplet` to specify the target triplet, e.g. `x64-windows` or `x86-windows`.
 
 ## Compile Luanti
 
-### a) Using the vcpkg toolchain and CMake GUI
+There are two ways to compile Luanti: CMake GUI and CLI
+
+### a) CMake GUI
 
-1. Start up the CMake GUI
-2. Select **Browse Source...** and select DIR/minetest
-3. Select **Browse Build...** and select DIR/minetest-build
+1. Start up the CMake GUI (Win > search "cmake-gui" > open)
+2. Select **Browse Source...** and select `DIR/minetest` (this is where you've cloned the repo)
+3. Select **Browse Build...** and select `DIR/minetest/build` (this is a new folder that CMake will prompt to create)
 4. Select **Configure**
-5. Choose the right visual Studio version and target platform. It has to match the version of the installed dependencies
+5. Choose the right Visual Studio version and target platform. Currently, Luanti uses Visual Studio 16 2019, but newer VS versions should work as well. The VS version has to match the version of the installed dependencies.
 6. Choose **Specify toolchain file for cross-compiling**
 7. Click **Next**
 8. Select the vcpkg toolchain file e.g. `D:/vcpkg/scripts/buildsystems/vcpkg.cmake`
 9. Click Finish
-10. Wait until cmake have generated the cash file
+10. Wait until CMake generates the cache file (this may take about 10-30 minutes, depending on your device)
 11. If there are any errors, solve them and hit **Configure**
 12. Click **Generate**
 13. Click **Open Project**
-14. Compile Luanti inside Visual studio.
+14. Compile Luanti inside Visual Studio.
+    - If you get "Unable to start program '...\x64\Debug\ALL_BUILD'. Access is denied", try compiling via the CLI instead.
 
-### b) Using the vcpkg toolchain and the commandline
+### b) CLI
 
 Run the following script in PowerShell:
 
 ```powershell
 cmake . -G"Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
 cmake --build . --config Release
 ```
```

Comment:
We should not be encouraging in-tree builds. Also, I think CMake should automatically pick the correct generator if it is left unspecified.

Reply:
Hmm, personally I don't understand this, happy to take action if there are any suggestions :)

Reply:
Running `cmake` from the build directory will cause CMake to output its build artifacts in the source tree. It's a pain to clean up, and it easily clutters output from tooling. It's better to run `cmake -B <name_of_build_directory>` which will limit the scope of CMake's shenanigans to single directory, which can be deleted in its entirely if it gets messed up later to start fresh. It also makes it easy to maintain multiple builds with different configurations.

Reply:
I would suggest to change it to
```
cmake -B build -G"Visual Studio 16 2019" -DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_CURSES=OFF
cmake --build build--config Release
```

I don't know how risky it would be to omit the `-G` option. It works on my machine because I use the CMake installed with the Visual Studio developer terminal, but I feel there's a risk that a different CMake installation might pick the wrong generator. I'm not well enough informed.

Reply:
I think we should keep the -G option for consistency

Reply:
However, users will need to update the value based on their installed version of VS, ref #15336

Reply:
Changed `.` to `build` :)